libsplinter/src/admin/store/diesel/operations/list_circuits.rs

```diff
@@ -231,6 +196,12 @@ where
                     })?);
                 }
 
+                ret_circuits.retain(|circuit| {
```

Comment:
This should still be done at the database level.  The correct solution should be to omit the filters from the query if it they are empty.

Reply:
Updated